neo/VM/ExecutionEngine.py

```diff
@@ -37,14 +37,17 @@ class ExecutionEngine():
     # file descriptor
     log_file = None
 
+    def is_write_log(self):
```

Comment:
I think one way to further speed things up determine on `__init__` a property called `_is_write_log` which is just the value of `settings.log_vm_instructions`.  

Then in line 912 you can just do `if self._is_write_log: ...` and in the `write_log` method you can check if the log file exists and if its not closed before actually writing.

Reply:
Sure thing; done!